sn_networking/src/replication_fetcher.rs

```diff
@@ -5,142 +5,195 @@
 // under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
+#![allow(clippy::mutable_key_type)]
```

Comment:
why do we need this?

Reply:
This is raised when the `RecordKey` is put inside `HashSet`. RecordKey is just a wrapper around `Bytes` and bytes has this field `data: AtomicPtr<()>` that can enable interior mutability. Think this is an open issue, https://github.com/rust-lang/rust-clippy/issues/5812